rpc/src/v1/impls/signing.rs

```diff
@@ -125,11 +125,24 @@ impl<D: Dispatcher + 'static> SigningQueueClient<D> {
 						.map(dispatch::WithToken::into_value)
 						.map(DispatchResult::Value))
 				} else {
-					Either::B(future::done(
-						signer.add_request(payload, origin)
-							.map(|(id, future)| DispatchResult::Future(id, future))
-							.map_err(|_| errors::request_rejected_limit())
-					))
+					if signer.is_enabled() {
```

Comment:
dumb q perhaps, but can't you do this check earlier?

Reply:
earlier in the function? ..

Reply:
yeah, like [here](https://github.com/paritytech/parity-ethereum/pull/11272/files#diff-ff06d97133c1cfeab870766a52830facR116)?

Reply:
yeah definitely